apps/src/templates/projects/projectsRedux.js

```diff
@@ -302,6 +343,57 @@ export function updateProjectName(projectId, updatedName) {
   return {type: UPDATE_PROJECT_NAME, projectId, updatedName};
 }
 
+export function saveSuccess(projectId) {
+  return {type: SAVE_SUCCESS, projectId};
+}
+
+export function saveFailure(projectId) {
+  return {type: SAVE_FAILURE, projectId};
+}
+
+const fetchProjectToUpdate = (projectId, onComplete) => {
+  $.ajax({
+    url: `/v3/channels/${projectId}`,
+    method: 'GET',
+    type: 'json',
+    contentType: 'application/json;charset=UTF-8',
+  }).done((data) => {
+    onComplete(null, data);
+  }).fail((jqXhr, status) => {
+    onComplete(status, jqXhr.responseJSON);
+  });
+};
+
+const updateProjectNameOnServer = (project) => {
+  return (dispatch) => {
+    $.ajax({
+      url: `/v3/channels/${project.id}`,
+      method: 'POST',
+      type: 'json',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify(project)
+    }).done((data) => {
+      dispatch(saveSuccess(project.id));
+    }).fail((jqXhr, status) => {
+      dispatch(saveFailure(project.id));
+    });
+  };
+};
+
+export const saveProjectName = (projectId, updatedName) => {
+  return (dispatch) => {
```

Comment:
Not required, but it might be a fun exercise to rewrite these methods using async/await so that `saveProjectName` collapses to:
```js
export function saveProjectName(projectId, updatedName) {
  return async (dispatch) => {
    try {
      const data = await fetchProjectToUpdate(projectId);
      data.name = updatedName;
      await updateProjectNameOnServer(data);
      dispatch(saveSuccess(projectId))
    } catch (err) {
      console.error(err);
      dispatch(saveFailure(projectId));
    }
  };
}
```